src/main/java/rx/Observable.java

```diff
@@ -8770,6 +8770,30 @@ public final Subscription subscribe(Subscriber<? super T> subscriber) {
     }
 
     /**
+     * Merges the specified observable sequence into this Observable sequence by using the resultSelector 
+     * function only when the source observable sequence (this instance) produces an element.
+     * <code><pre>
+     * ----A-------B------C----->  o1
+     *
+     * --0----1-2----3-4-------->  o2
+     *
+     *     |       |      |
+     *     V       V      V
+     *
+     *   (A,0)   (B,2)  (C,4)
+     * </pre></code>
+     * @param other the other observable sequence
+     * @param resultSelector the function to call when this Observable emits an element and the other
```

Comment:
Naming this "resultSelector" seems off (copy/paste?) since it's more akin to `combineLatest` which uses "combineFunction".